src/ndarray/ndarray_function-inl.h

```diff
@@ -433,6 +442,27 @@ void EvalBroadcast<DEVICE>(TBlob const& src, TBlob* ret, int size, RunContext ct
   out = mshadow::expr::broadcast_with_axis(in, 0, size);
 }
 
+/*! \brief Minimum of three */
+MSHADOW_XINLINE size_t minthree(const size_t a, const size_t b, const size_t c) {
+  return a < b ? (a < c ? a : c) : (b < c ? b : c);
+}
+
+template<typename OP, typename xpu>
+void BinaryOpKernelLaunch(mshadow::Stream<xpu>* s, const TBlob& lhs, const TBlob& rhs, TBlob *out) {
+  using namespace op::mxnet_op;
+  using namespace mshadow;
+  MSHADOW_TYPE_SWITCH(out->type_flag_, DType, {
+    const size_t size = (minthree(out->Size(), lhs.Size(), rhs.Size())
+                         + DataType<DType>::kLanes - 1)
+                        / DataType<DType>::kLanes;
```

Comment:
why do you need to calc size like this?

Reply:
You're right, at this point lhs, rhs and out should be same size. So we don't need this. Removed